GameSettings: Various Nick games fixes

icup321 opened the pull request:
This PR provides a batch of GameSettings updates:
- Sets SafeTextureCacheColorSamples to proper values to fix choppy/pixelated FMVs.
- Disable texture filtering for several games to prevent broken FMVs.
- Disable ImmediateXFBEnable to prevent some FMV hangs in Nicktoons: Battle for Volcano Island and Barnyard Wii specifically.

JMC47 commented:
These look good at a glance, most of them make sense based on what I know of these games.

I do prefer that we actually have comments in the INI to say why a setting is chosen, especially now that graphics settings are available directly in the UI.

Overall though I think these are correct and it shouldn't be blocking.

icup321 commented:
I've went ahead and added comments to all of the FMV fixes as requested. I also added more fixes for other Nickelodeon games that I was originally going to save for a separate PR, but decided to just merge them all into one.
